ssntp/ssntp.go

```diff
@@ -1151,13 +1155,76 @@ func (config *Config) port() uint32 {
 	return port
 }
 
-func (config *Config) setCerts() {
-	if config.CAcert == "" {
-		config.CAcert = DefaultCACert
+func loadCertificate(certPath string) (*x509.Certificate, error) {
+	certPEM, err := ioutil.ReadFile(certPath)
+	if err != nil {
+		return nil, err
+	}
+	block, _ := pem.Decode(certPEM)
+	if block == nil {
+		panic("failed to parse certificate PEM")
+	}
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse certificate: %v", err)
+	}
+	return cert, nil
+}
+
+func getDefaultCertificate() (cacert, cert string, err error) {
+	certs := []string{}
+
+	files, err := filepath.Glob(ciaoCertsPath)
+	if err != nil {
+		fmt.Println(err)
+		return "", "", err
 	}
 
-	if config.Cert == "" {
-		config.Cert = defaultClientCert
+certsLoop:
+	for _, file := range files {
+		if cacert == "" {
+			cert, err := loadCertificate(file)
+			if err != nil {
+				return "", "", err
+			}
+			if cert.IsCA == true {
+				cacert = file
+				continue certsLoop
+			}
+		}
+		certs = append(certs, file)
+	}
+
+	if len(certs) > 1 {
+		return "", "", fmt.Errorf("More than one cert files at: %s", ciaoCertsPath)
+	} else if len(certs) == 1 {
+		certPEM, err := ioutil.ReadFile(cacert)
```

Comment:
The code in this block represents the success condition and so should be on the main path through the function, e.g.,

```
} else if len(certs) == 0 {
    return "", "", fmt.Errorf("%s Certificates are not found", ciaoCertsPath)
}

certPEM, err := ioutil.ReadFile(cacert)
...